### PR TITLE
add regularizer for convlstmpeephole3d

### PR DESCRIPTION
**Pull request by dding3**
## What changes were proposed in this pull request?
1. add regularizer for convlstmpeephole3d
2. add more documents for recurrent
3. rename convlstm2d back to convlstm

## How was this patch tested?
ut

### CODE REVIEW

File: docs/docs/APIdocs/Layers/Recurrent-Layers.md [hunk 1]
```diff
@@ -9,7 +9,15 @@ val module = Recurrent()
 module = Recurrent()
 ```
 
-Recurrent module is a container of rnn cells. Different types of rnn cells can be added using add() function.
+Recurrent module is a container of rnn cells. Different types of rnn cells can be added using add() function.  
+
+Recurrent support return final state and cell status of its rnn cells by using getFinalStateAndCellStatus. output of getFinalStateAndCellStatus
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/Layers/Recurrent-Layers.md]: Recurrent supports returning?

Comment (dding3) [inline docs/docs/APIdocs/Layers/Recurrent-Layers.md]: I have added getFinalStateAndCellStatus in recurrent, which support return. Code has been merged, this pr adds some explaination

Comment (SeaOfOcean) [inline docs/docs/APIdocs/Layers/Recurrent-Layers.md]: I mean the syntax

**Comment by SeaOfOcean** [comment]
Also add regularizer for CMul

**Comment by SeaOfOcean** [comment]
add a unit test for convlstmpeephole3d with regularizer

**Comment by SeaOfOcean** [comment]
LGTM

**Pull request merged**